core/consensus/src/engine.rs

```diff
@@ -98,6 +100,10 @@ impl<Adapter: ConsensusAdapter + 'static> Engine<Proposal> for ConsensusEngine<A
             }
         };
 
+        if remove {
+            self.adapter.remove_hardfork_proposal(ctx.clone()).await?;
```

Comment:
It should be removed after the execution of the transactions in this block is complete.

Reply:
This proposal has expired. In this case, it can only be deleted. It is impossible to propose again.

Reply:
Not sure if it is proper to change the state in a `get` function (get_block)?